cvat-ui/src/actions/models-actions.ts

```diff
@@ -64,34 +91,87 @@ export const modelsActions = {
             taskInstance,
         })
     ),
+    getModelProviders: () => createAction(ModelsActionTypes.GET_MODEL_PROVIDERS),
+    getModelProvidersSuccess: (providers: ModelProvider[]) => (
+        createAction(ModelsActionTypes.GET_MODEL_PROVIDERS_SUCCESS, {
+            providers,
+        })),
+    getModelProvidersFailed: (error: any) => createAction(ModelsActionTypes.GET_MODEL_PROVIDERS_FAILED, { error }),
+    getModelPreview: (modelID: string) => (
+        createAction(ModelsActionTypes.GET_MODEL_PREVIEW, { modelID })
+    ),
+    getModelPreviewSuccess: (modelID: string, preview: string) => (
+        createAction(ModelsActionTypes.GET_MODEL_PREVIEW_SUCCESS, { modelID, preview })
+    ),
+    getModelPreviewFailed: (modelID: string, error: any) => (
+        createAction(ModelsActionTypes.GET_MODEL_PREVIEW_FAILED, { modelID, error })
+    ),
 };
 
 export type ModelsActions = ActionUnion<typeof modelsActions>;
 
 const core = getCore();
 
-export function getModelsAsync(): ThunkAction {
+export function getModelsAsync(query?: ModelsQuery): ThunkAction {
     return async (dispatch): Promise<void> => {
-        dispatch(modelsActions.getModels());
+        dispatch(modelsActions.getModels(query));
+
+        const filteredQuery = { ...query };
+        if (query) {
+            for (const key of Object.keys(query)) {
+                if ((filteredQuery as Indexable)[key] === null) {
+                    delete (filteredQuery as Indexable)[key];
+                }
+            }
+        }
 
         try {
-            const models = await core.lambda.list();
+            const models = await core.lambda.list(filteredQuery);
             dispatch(modelsActions.getModelsSuccess(models));
         } catch (error) {
             dispatch(modelsActions.getModelsFailed(error));
         }
     };
 }
 
+export function createModelAsync(modelData: Record<string, string>): ThunkAction {
+    return async function (dispatch) {
+        const model = new cvat.classes.MLModel(modelData);
+
+        dispatch(modelsActions.createModel());
+        try {
+            const createdModel = await model.save();
+            dispatch(modelsActions.createModelSuccess(createdModel));
+        } catch (error) {
+            dispatch(modelsActions.createModelFailed(error));
+            throw error;
```

Comment:
What is the reason to throw the error after dispatching error action?

Reply:
We need it in create model form so we can catch it, and success message is not displayed